test/validation/test_formatting_validation.py

```diff
@@ -242,6 +243,23 @@ def test_check_for_fenced_code_block(test_input: str, should_match: bool) -> Non
     assert actual == expected
 
 
+@pytest.mark.parametrize(
+    "test_input,should_match",
+    [
+        ("—-", True),
+        ("-—", True),
+        ("---------", False),
+        ("---"), False),
+        ("Word\n    int x = 1\nWord", False),
+    ],
+)
+def test_check_for_autogenerated_long_hyphen(test_input: str, should_match: bool) -> None:
+    """Test if autogenerated long hyphens are detected."""

```

Comment:
The long hyphen isn't always auto-generated, so perhaps "accidental" would be a better descriptor than "autogenerated"?

Side note: if settings for "smart quotes" (and the like) are enabled on macOS, it automatically replaces `--` with `—` there too. Not just mobile. 😄